language/move-vm/runtime/src/runtime.rs

```diff
@@ -124,7 +124,16 @@ impl VMRuntime {
                 .map(Value::transaction_argument_signer_reference)
                 .collect()
         } else {
-            // no signer parameters, don't add to args
+            // no signer parameters
+            if false
+            /* senders.is_empty() */
```

Comment:
This was a crutch for migrating to `signer`s from the `move_to_sender` world, since we had no way of specifying what the signer should be, the Move VM had to infer it.

Eventually we could totally move from "all or nothing" to "all specified" for signers, but this would require adding a new argument structure to the transaction type. 

So I guess my point is I don't see the value in this PR right now since its a known problem (and this code isn't run :p) 
If we want to leave it here as a fancy TODO, I suppose that is okay

Reply:
> Eventually we could totally move from "all or nothing" to "all specified" for signers, but this would require adding a new argument structure to the transaction type.

I actually don't think we would need to change this as long as we're ok with every Libra tx script having >0 `&signer`s (which seems like a reasonable restriction).

Reply:
I'm ok with keeping or removing the dead code (perhaps better to kill the code + keep the comment), just wanted to demonstrate the fix.

Reply:
> I actually don't think we would need to change this as long as we're ok with every Libra tx script having >0 &signers (which seems like a reasonable restriction).

I was going to speak to this but deleted the comment out of sadness. Its a lot of work to do that at the moment without a lot of time... I'd be on board for the change, but just not sure if it is worth it at this point

Reply:
Agreed. Definitely not high-pri to fix (especially given the volume of work), but perhaps good to keep in mind as an item to make progress on as we're refactoring the tests.